src/Pate/Proof/Instances.hs

```diff
@@ -641,11 +642,11 @@ cellInGroundDomain dom cell = case isStackCell cell of
   True -> cellInMemDomain (PED.eqDomainStackMemory dom) cell
   False -> cellInMemDomain (PED.eqDomainGlobalMemory dom) cell
 
-ppExitCase :: MS.MacawBlockEndCase -> String
+ppExitCase :: MCS.MacawBlockEndCase -> String
 ppExitCase ec = case ec of
-  MS.MacawBlockEndJump -> "arbitrary jump"
-  MS.MacawBlockEndCall -> "function call"
-  MS.MacawBlockEndReturn -> "function return"
-  MS.MacawBlockEndBranch -> "branch"
-  MS.MacawBlockEndArch -> "syscall"
-  MS.MacawBlockEndFail -> "analysis failure"
+  MCS.MacawBlockEndJump -> "arbitrary jump"
+  MCS.MacawBlockEndCall -> "function call"
+  MCS.MacawBlockEndReturn -> "function return"
+  MCS.MacawBlockEndBranch -> "branch"
+  MCS.MacawBlockEndArch -> "arch-specific"
```

Comment:
Since we have a typeclass here, can we now make a more informative prettyprinter here?